File: app/src/main/java/com/nollpointer/dates/sort/SortCardView.java
package com.nollpointer.dates.sort;

import androidx.cardview.widget.CardView;
import android.view.View;
import android.widget.TextView;

import com.nollpointer.dates.R;

public class SortCardView {
    CardView card;

    TextView number;
    TextView event;

    SortCardView(CardView card) {
        this.card = card;
        number = card.findViewById(R.id.text1);
        event = card.findViewById(R.id.text2);
    }

    public void setBackgroundColor(int color) {
        card.setCardBackgroundColor(color);
    }

    public void setOnClickListener(View.OnClickListener listener) {
        card.setOnClickListener(listener);
    }

    public void setEvent(String text) {
        event.setText(text);
    }

    public void setNumber(int number) {
        this.number.setText(Integer.toString(number));
    }

    //public void setResultScreenText(String)


}

File: app/src/main/java/com/nollpointer/dates/ui/sort/SortCardView.kt
package com.nollpointer.dates.ui.sort

import android.view.View
import android.widget.TextView
import androidx.cardview.widget.CardView
import com.nollpointer.dates.R

class SortCardView internal constructor(private val card: CardView) {

    private val number: TextView = card.findViewById(R.id.text1)
    private val event: TextView = card.findViewById(R.id.text2)

    fun setBackgroundColor(color: Int) {
        card.setCardBackgroundColor(color)
    }

    fun setOnClickListener(listener: View.OnClickListener?) {
        card.setOnClickListener(listener)
    }

    fun setEvent(text: String?) {
        event.text = text
    }

    fun setNumber(number: Int) {
        this.number.text = number.toString()
    }

    //public void setResultScreenText(String)

}